### ISSUE TITLE
ParameterValidationDataFlowOperationVisitor asserts

### ISSUE THREAD

**Issue by dotpaul** [dotnet/roslyn-analyzers#3518]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (latest release), debug bits

#### Diagnostic ID

CA1062

#### Repro steps

Add this unit test to 

```csharp
        [Fact]
        public void Blah()
        {
            VerifyCSharp(@"
using System;

public class Class1
{
    private delegate void ActionWrapper();

    public static void DoSomething<P1>(Action action)
    {
        // Ensure no diagnostic here.
        Bar2(action.Invoke);
    }

    private static void Bar2(ActionWrapper wrapper)
    {
        wrapper();
    }
}");
        }
```

#### Expected behavior
No asserts.

#### Actual behavior
`Debug.Assert(!targetMethod.IsLambdaOrLocalFunctionOrDelegate())` fires.
